src/Maybe.js

```diff
@@ -1,45 +1,82 @@
-module.exports = Maybe;
+var util = require('./internal/util.js');
 
 function Maybe(x) {
-    if (!(this instanceof Maybe)) {
-        return new Maybe(x);
-    }
+    throw new Error('Maybe cannot be directly instantiated.');
+};
+
+function _Just(x) {
     this.value = x;
+}
+util.extend(_Just, Maybe);
+
+function _Nothing() {}
+util.extend(_Nothing, Maybe);
+
+var _nothing = new _Nothing();
+
+Maybe.Nothing = function() {
+    return _nothing;
 };
 
-Maybe.of = function(x) {
-    return new Maybe(x);
+Maybe.Just = function(x) {
+    return new _Just(x);
 };
 
+Maybe.from = function(x) {
```

Comment:
Apparently ```from``` is a method on ```Comonad``` https://github.com/fantasyland/fantasy-land/issues/85#issuecomment-73523980